safehttp/plugins/csp/csp.go

```diff
@@ -157,11 +160,31 @@ type FramingPolicy struct {
 // to Serialize which can be used in 'nonce-{random-nonce}' values in directives.
 func (f FramingPolicy) Serialize(nonce string) string {
 	var b strings.Builder
+
+	b.WriteString(frameAncestors(f.Sources))
+	b.WriteString(report(f.ReportURI))
+
+	return b.String()
+}
+
+func frameAncestors(sources []string) string {
+	var b strings.Builder
 	b.WriteString("frame-ancestors 'self'")
 
-	if f.ReportURI != "" {
+	for _, s := range sources {
+		b.WriteString(" ")
+		b.WriteString(s)
+	}
+
+	return b.String()
+}
+
+func report(reportURI string) string {
+	var b strings.Builder
+
+	if reportURI != "" {
 		b.WriteString("; report-uri ")
```

Comment:
This is brittle.

There is no reason for `Serialize` to not write `report()` first and then `frameAncestors`, but putting the semicolon at the beginning breaks this.

Reply:
Ok, I made it order agnostic.